dotnet/src/Experimental/Orchestration.Flow.IntegrationTests/CollectEmailPlugin.cs

```diff
@@ -47,29 +46,33 @@ public CollectEmailPlugin(Kernel kernel)
     [Description("Useful to assist in configuration of email address, must be called after email provided")]
     public async Task<string> CollectEmailAsync(
         [Description("The email address provided by the user, pass no matter what the value is")]
+        // ReSharper disable once InconsistentNaming
+#pragma warning disable CA1707 // Identifiers should not contain underscores
         string email_address,
-        ContextVariables variables)
+#pragma warning restore CA1707 // Identifiers should not contain underscores
+        KernelArguments arguments)
     {
-        var chat = this._chat.CreateNewChat(SystemPrompt);
+        var chat = new ChatHistory(SystemPrompt);
         chat.AddUserMessage(Goal);
 
-        ChatHistory? chatHistory = variables.GetChatHistory();
-        if (chatHistory?.Any() ?? false)
+        ChatHistory? chatHistory = arguments.GetChatHistory();
```

Comment:
It seems that the only thing needed for the function from the arguments dictionary is the chat history. If that is the case, consider passing chat history directly through the method parameter: CollectEmailAsync(string emailAddress, ChatHistory chatHistory).

Reply:
`ChatHistory` is usually large and structured, which is difficult for ReAct engine to infer and pass.
instead of making it a required parameter for each semantic function, right now it is designed to use a more implicit and on-demand pattern.